internal/operators/cnv/cnv_operator.go

```diff
@@ -70,11 +70,16 @@ func (o *operator) ValidateCluster(_ context.Context, _ *common.Cluster) (api.Va
 // ValidateHost returns validationResult based on node type requirements such as memory and cpu
 func (o *operator) ValidateHost(ctx context.Context, cluster *common.Cluster, host *models.Host) (api.ValidationResult, error) {
 	var inventory models.Inventory
+	if host.Inventory == "" {
+		o.log.Info("Empty Inventory of host with hostID ", host.ID)
+		return api.ValidationResult{Status: api.Pending, ValidationId: o.GetClusterValidationID(), Reasons: []string{"Missing Inventory in some of the hosts"}}, nil
+	}
 	if err := json.Unmarshal([]byte(host.Inventory), &inventory); err != nil {
 		o.log.Errorf("Failed to get inventory from host with id %s", host.ID)
 		return api.ValidationResult{Status: api.Failure, ValidationId: o.GetClusterValidationID()}, err
 	}
 	// TODO: validate available devices on worker node like gpu and sr-iov and check whether there is enough memory to support them
+	// TODO: how to handle validation for model.HostRoleAutoAssign ???
```

Comment:
@gamli75 @YuviGold This seems to prevent host validation. The role flips as we run consecutive reruns of `RefreshStatus` for specific host. This makes the validation responses to be overridden and it makes it impossible to write validation end to end tests. How do you propose to change this?

Reply:
Could we explicitly set a host role before we check the validation on the e2e tests?

Regarding how it should work, I see that OCS validation fails in case of a role wasn't [assigned](https://github.com/openshift/assisted-service/blob/5c64d7b843e9d024826d172cfd02545b8a889813/internal/operators/ocs/validations.go#L82-L90)
On the other hand, other host validations such as [CPUs amount](https://github.com/openshift/assisted-service/blob/709ddcfd8921a44980c780192f4815fdb6a88546/internal/host/validator.go#L280) and [sufficient memory](https://github.com/openshift/assisted-service/blob/709ddcfd8921a44980c780192f4815fdb6a88546/internal/host/validator.go#L316) treats auto-assign as a worker.
I'm not sure what is the better approach - but I think auto-assign role handling should be aligned. WDYT? @avishayt @filanov

Reply:
> Could we explicitly set a host role before we check the validation on the e2e tests?
> 

As far as I understand we can only tell how many masters and nodes. I am not sure how to assign a role upfront.

> Regarding how it should work, I see that OCS validation fails in case of a role wasn't [assigned](https://github.com/openshift/assisted-service/blob/5c64d7b843e9d024826d172cfd02545b8a889813/internal/operators/ocs/validations.go#L82-L90)

I know that OCS perform only cluster validation not per host validation. I can fail validation in similar way but it sound misleading. 

> On the other hand, other host validations such as [CPUs amount](https://github.com/openshift/assisted-service/blob/709ddcfd8921a44980c780192f4815fdb6a88546/internal/host/validator.go#L280) and [sufficient memory](https://github.com/openshift/assisted-service/blob/709ddcfd8921a44980c780192f4815fdb6a88546/internal/host/validator.go#L316) treats auto-assign as a worker.

This is not correct with behavior I observed. I see hosts flapping the role from master to autoassigned and back. I have not seen worker role to be assigned at all.

Reply:
Last week we talked about maybe enforcing the user to set masters, i can't find the thread for some reason. @avishayt do you remember what was the result of that conversation?

Reply:
> Could we explicitly set a host role before we check the validation on the e2e tests?

@pkliczewski You can use the UpdateCluster API - 
https://github.com/openshift/assisted-service/blob/df6988c73f2f3a888fc199eec84d1fd3940d131e/swagger.yaml#L3691-L3702

Reply:
I will use this in end to end which should help me to create tests. I am not sure whether it would help to solve anything in production assuming that roles are not preassigned.

Reply:
My recommendation is to align with OCS (fail the validation if there are more than 3 hosts and at least 1 is auto-assign).  Once we fix the auto-assign feature we can change the behavior.

Reply:
OCS uses `ValidateCluster` where they know how many hosts there are in CNV case we use `ValidateHost` where we check only one host at a time. Behavior consistency make sense but in principal it doesn't.